session.go

```diff
@@ -645,12 +645,21 @@ func (s *Session) mux(remoteBegin *frames.PerformBegin) {
 			}
 
 			s.txFrame(env.Ctx, fr, env.Sent)
+
 			if sendErr := <-env.Sent; sendErr != nil {
 				s.doneErr = sendErr
 
 				// put the error back as our sender will read from this channel
 				env.Sent <- sendErr
-				return
+
+				if errors.Is(sendErr, context.Canceled) || errors.Is(sendErr, context.DeadlineExceeded) {
```

Comment:
This shouldn't be necessary.  Receiving a `*ConnError` means that the conn has died, so on the next iteration we hit this block.
```go
// conn has completed, exit
case <-s.conn.done:
	s.doneErr = s.conn.doneErr
	return
```